interpreter/interpreter.go

```diff
@@ -892,19 +664,33 @@ type Value interface {
 
 type function struct {
 	e     *semantic.FunctionExpression
-	scope *Scope
+	scope Scope
 
-	typ      semantic.Type
-	polyType semantic.PolyType
+	types     map[semantic.Node]semantic.Type
+	polyTypes map[semantic.Node]semantic.PolyType
```

Comment:
Are these basically a cache of all child nodes of the function expression so that later when we need to look up types we have a copy?

Reply:
Yes this is in place of the former TypeScope. This is just temporary and will be removed eventually.